src/app/core/model/hal/pattern-container.model.ts

```diff
@@ -0,0 +1,15 @@
+import Pattern from './pattern.model';
+import UriEntity from './uri-entity.model';
+import { HalLink } from './hal-link.interface';
+
+export class PatternContainer extends UriEntity {
```

Comment:
This entity used to be called PatternViews, not it is a patternContainer. 
What is a pattern container? Is it an abstraction about pattern views and your new entities for the new module and implies that there is always a graph? 
Is a pattern language a pattern container? 
I think a few sentences explaining your new concept/abstraciton would be great

Reply:
For me this class looked like an DTO, which is also implied by the `hal` folder and as instances of it include an array of patterns and some links.

As this class is used directly in the GraphDisplayComponent, there are three options:
1. add another condition everywhere this PatternViews is used in the component
2. do not use DTOs directly in components (this would may be the best solution, with most effort)
3. abstract it and use one array wrapper for pattern views and design models (I have chosen this option, as trade off)

Reply:
@marzn could you do me a favor and add a comment in the code that explains what this entity is for? Just in case I forget it, or another student comes across it.
patternContainer is a really generic name that (especially in a pattern repository project) could mean multiple things

Reply:
Sure, I can do that. Or if you prefer another name, it can simply be changed.